### DISCUSSION TITLE
Simplest setup failing with 2035' ('MQRC_NOT_AUTHORIZED').

### DISCUSSION THREAD

**Discussion by fatso83** [ibm-messaging/mq-container#563] [General]
Just as in https://github.com/ibm-messaging/mq-container/discussions/496, I fail trying to connect using the default settings. I have a Spring Boot application setup to use these settings:

## App Config
```yaml
ibm:
  mq:
    queue-manager: QM1
    channel: DEV.APP.SVRCONN
    conn-name: localhost(1414)
    queue: DEV.QUEUE.1
    user: "app"
    password: "my-app-pass" # I also tried with a blank pass to begin with, same error
```

## How I start it
```
export MQ_APP_PASSWORD=my-app-pass # also tried not setting this
docker run   --env LICENSE=accept   --env MQ_QMGR_NAME=QM1   \
        --publish 1414:1414    \
        --publish 9443:9443      \
        ibm-mqadvanced-server-dev:9.3.5.1-arm64
```

## Console output
```
29.05.2024 23:51:57,656 [CID:]  [RMI TCP Connection(4)-127.0.0.1] WARN  [actuate.jms.JmsHealthIndicator] - JMS health check failed
com.ibm.msg.client.jakarta.jms.DetailedJMSSecurityException: JMSWMQ2013: The security authentication was not valid that was supplied for queue manager 'QM1' with connection mode 'Client' and host name 'localhost(1414)'.
...
Caused by: com.ibm.mq.MQException: JMSCMQ0001: IBM MQ call failed with compcode '2' ('MQCC_FAILED') reason '2035' ('MQRC_NOT_AUTHORIZED').
	at com.ibm.msg.client.jakarta.wmq.common.internal.Reason.createException(Reason.java:203) ~[com.ibm.mq.jakarta.client-9.3.5.0.jar:9.3.5.0 - p935-L240201]
	... 55 more
```

## Env info
Obviously (from the docker image name) you can see that I'm running ARM64, if it should matter.

So obviously, something related to authentication or authorization is happening, unsure what. I just try to stick to the default settings.

**Comment by fatso83** [comment]
I also tried disabling security, but then it failed due to supplying an empty string as a password:
```
2024-05-29T22:03:50.961Z AMQ9791E: The client application did not supply a user ID and password. [CommentInsert1(batch.Application), CommentInsert2(192.168.65.1), CommentInsert3(MCAUSER(app) CLNTUSER(app))]
2024-05-29T22:03:50.961Z AMQ9999E: Channel 'DEV.APP.SVRCONN' to host '192.168.65.1' ended abnormally. [CommentInsert1(DEV.APP.SVRCONN), CommentInsert2(635), CommentInsert3(192.168.65.1)]

```

No go.

**Comment by arthurbarr** [comment]
The behaviour has changed so that now you must set a password for the user, otherwise the user is not available.  See [Environment variables for the default developer config](https://github.com/ibm-messaging/mq-container/blob/master/docs/developer-config.md#environment-variables).

**Reply by fatso83** [reply]
But I did set the password? You can see that in the top, as I export it. Same thing if inline:
```
MQ_APP_PASSWORD=my-app-password docker run   --env LICENSE=accept   --env MQ_QMGR_NAME=QM1   --publish 1414:1414   --publish 9443:9443      ibm-mqadvanced-server-dev:9.3.5.1-arm64
```

I still get the same error. The password > 8 characters. I tried stepping through the code, but there is something wrong with the ibm-mq libs  uploaded to Maven Central, so IntelliJ fails when doing it. I did see the password and username being set on the application side, though (in `com.ibm.mq.spring.boot.MQConfigurationProperties`).

**Comment by arthurbarr** [comment]
You need to use the [`--env`](https://docs.docker.com/reference/cli/docker/container/run/#env) option to `docker run` to set the environment variable inside the container.  Part of the point of running containers is to isolate them from the environment of other processes, so the environment of your shell is not passed into the container unless you explicitly request it.

**Reply by fatso83** [reply]
Haha, damn, I knew there was something I was overlooking (hence choosing discussion over issue). Thanks a lot!

<img src="https://media3.giphy.com/media/e5uyWolyR0y30Wo1ya/giphy.gif"/>

I'll post the full working incantation here for posterity:
```
docker run --env MQ_APP_PASSWORD=my-app-password   --env LICENSE=accept   --env MQ_QMGR_NAME=QM1   --publish 1414:1414   --publish 9443:9443      ibm-mqadvanced-server-dev:9.3.5.1-arm64   
```
